src/dataframe/dataframe.jl

```diff
@@ -802,14 +809,8 @@ function Base.append!(df1::DataFrame, df2::AbstractDataFrame)
    return df1
 end
 
-function Base.convert(::Type{DataFrame}, A::AbstractMatrix)
-    n = size(A, 2)
-    cols = Vector{Any}(n)
-    for i in 1:n
-        cols[i] = A[:, i]
-    end
-    return DataFrame(cols, Index(gennames(n)))
-end
+Base.convert(::Type{DataFrame}, A::AbstractMatrix) = DataFrame(A)
+Base.convert(::Type{DataFrame}, V::AbstractVector) = DataFrame(V)
```

Comment:
Note that this method is very different from the previous one as it expects the entries to be columns. I'm not sure we should support it, as it could be confusing compared with a 1-column matrix.

Reply:
To say the truth I do not like `DataFrame(::Vector)` constructor, especially something like `DataFrame([1,2,3])` is very non-intuivite. I pass a column vector and get a row of data.

But if we allow it then it is consistent to provide `convert` also.

Maybe a solution is to restrict the implementation of `DataFrame(columns::AbstractVector` to check internally:
```
all(col -> isa(col, AbstractVector), columns) || throw(ArgumentError("columns must be vectors"))
```

Reply:
Yeah, it's a bit weird. Maybe we could also require specifying the column names when passing a vector, given that this is a rather low-level constructor.

Reply:
agreed. this means that `convert` should be removed. I add deprecation to calling *vector of vectors* `DataFrame` without passing names.